### DISCUSSION TITLE
Thank you Amazon Lumberyard devs

### DISCUSSION THREAD

**Discussion by ibex-are-goats** [o3de/o3de#1872] [General]
I can tell from the mixed language in the documentation that this change from Amazon Lumberyard to Linux Foundation O3DE was a startling one. I'm sorry for anyone who will be displaced by this change, or who feels like their work wasn't respected. I hope that the ambition of giving the world a great new piece of technology comes true.

Regardless of what Amazon does in the future, I want to thank the devs from Amazon Lumberyard for their work.

**Comment by audiojak** [comment]
Hi! Sorry if the language is not clear yet. This was well planned. In fact, we have been working on the new version for over a year.

**Reply by kberg0** [reply]
Well over a year in fact. What you are seeing are the results of us prepping the engine and documentation before our actual license was changed to Apache 2.0. If anything had leaked prior to the license change it could have been quite damaging to Amazon and our partners. The documentation team engaged in a monumental effort crunching over this last weekend especially to validate that all the license headers and references in code and in documentation had been correctly updated.

The whole team at Amazon working on O3DE is still very much here, and will continue to contribute to the engine for years to come.

**Reply by PreyK** [reply]
Do you tho?
I mean absolutely no personal disrespect to anyone and as a dev it's quite remarkable what could you do with the ancient CrySDK codebase (Lumberyard) but why do I feel like as soon as Amazon realizes their ROI on this will be "below expectations" this gets abandoned?

**Reply by OBWANDO** [reply]
Think of it in this light. It's not always about the immediate ROI. It's about long-term goals, and this is one of them. 

I understand that in some cases there is an urgency to make money immediately, but like many of the services people use every day, they were not profitable for some time while executing their vision. This was well understood going into O3DE as the vision had to have a long-term goal. 

That said, and to be more transparent, the more community adoption, the better the opportunity for all. 
It's just going to take time and support to get there.

**Reply by PreyK** [reply]
Yes, that’s absolultely right!
I’m trying to think of this in a different light but, Im just not sure they realize how much it will take to achive the scope of the project.
In the announcement video they talked about industry disruption, one open standard, I’v read about eventually an open 3D metaverse etc..
Theese are really bold claims and i was just wondering about who’s gonna do it you know?
The Linux Foundation is cool, but their essential sole porpuse is to make Linux.
The other partner companies (Adobe, niantic etc..) have very little experience in open-source and AAA game engine development.
The community can be powerful but right now this project’s only big advantage compared to other open source solutions out there is that Amazon backs it.
If they abandon this it’s gone and why would the community keep working on it when they have other community based projects that are more production ready?
I mean just take a look at Unity who are even a for profit solution, have immense resources, some of the best talent, many years of experience yet they are still struggling to deliver their on their vision for years.
Industry disruption means that I as a game developer will use O3DE instead of Unreal or Unity because it’s just better and more widely adopted.
Engines are hard. Companies went bankrupt on this, even Autodesk who could’v had it easily on paper pulled out (Stingray).
This is what i mean on ROI, that this really requies immense resources, and as i see it there’s almost a very little chance that it will be worth it for a for profit company even in the long run.
I’d love to be proven wrong tho, if this succeeds it’s gonna be huge and open source solutions would be very welcome in a space where essentially a few comapanies hold a monopoly over the whole industry.

**Reply by amznestebanpapp** [reply]
[NOTE] This is my personal view of Amazon's interest.

IMO Amazon's interest in the engine comes from multiple places:
- There is a huge amount of internal projects at Amazon that need a 3D Engine: games, warehouse simulations, robot training, studio in the cloud, etc.  You will be surprised how many internal projects at Amazon could use this tech.
- Lots of AWS customers are game companies. Game engines is a risky and expensive part of any game company. By giving them an engine, we make those game companies more likely to succeed which drives more AWS usage.

Making this a successful open source engine benefits Amazon in those two places (and others). Amazon will get a more feature-rich, stable and widely-adopted engine. This means things like:
- plugins supported by other companies, e.g. Amazon was supporting the WWise plugin, AudioKinetic is now a partner, they likely will take over that.
- more integrations to other tools, e.g. Lumberyard used to have plugins for Maya and 3DStudio, Adobe is now a partner, its more likely our asset ingestion pipeline will work better with those tools
- more platforms, e.g. Lumberyard did not have enough resources to support so many platforms. RedHat is now a partner, its more likely Linux will be more complete. Another example, there is already someone making Linux ARM work, this means that we could start running server instances in EC2 Graviton.
- Amazon can focus less in the core engine part and focus on AWS integrations and other technologies. However, dont take it as "Amazon will stop investing in the core engine", we need to keep investing there to make this a success.
- There are already talks about WebGL, which could benefit the "Studio in the cloud" business (https://aws.amazon.com/nimble-studio/)

Amazon will get a lot back from open sourcing it. To make that happen, it also has to invest in adoption, community and the engine itself. Any open source project that doesn't evolve and improve will eventually die. That's why we improved so many areas of the engine and continue to do so, we are actively investing in the engine.

Amazon's interest in the engine is not fading, all the contrary, the amount of changes been made is evidence is stronger than ever. My realization time ago was that Amazon is not interested in "making money" out of the engine. Its interested in driving other business and solving problems in other spaces where an engine is needed.
This is the same problem a lot of our customers have. Its a no-brainer when we can solve our problem and our customer's problems at the same time.

**Reply by kberg0** [reply]
If it's any reassurance, the active pull requests are right there for you or anyone to audit to confirm the accuracy of that claim. Even if Amazon did decide to pull the plug some years from now, many of the people working on this are so passionate about it that I'm very certain they would continue pushing this project forward on their own time.

**Reply by PreyK** [reply]
Thanks a lot for clarifying!
Solving problems for yourself then open-sourcing it for others is actually a really cool mentality, makes sense, and thanks for that!
I just couldn't wrap my head around this (Amazon with an open-source engine?) from the press releases and announcements so I thought I'd just poke around.
Switching engines can be a huge risk as a developer, wanted to get as clear of a picture of the project as I can before considering anything or becoming excited.
I actually looked this through today (looked at some of the PR-s too) have to say I'm impressed so far.
The technicals look great and cautiously optimistic about the Atom renderer too (AZSL and shader cross-compilation sound great, the modular rendering architecture makes sense etc..) need to look into it more but definitely looks solid! 
I might have misjudged Amazon here, was really excited about Lumberyard for years, and at first, this just sounded like "Lumberyard is dead, let's try to salvage the ashes" but yes, after looking through this and with the insights you guys gave this is definitely not that, sounds more like evolution and the next step.
Thanks again for taking the time to answer my concerns.
